pandas/plotting/_matplotlib/core.py

```diff
@@ -511,10 +523,24 @@ def _adorn_subplots(self):
             if self.xticks is not None:
                 ax.set_xticks(self.xticks)
 
+            # Addressing issue #40781 raising ValueError
+            # if xlim or ylim contain non float or non int dtype
             if self.ylim is not None:
+                if not self._has_iterable_only_float_int_or_datetime(self.ylim):
+                    raise ValueError(
```

Comment:
change the routine to _validate_valid_dtype and just raise inside

Reply:
working on it currently, should fix soon.

Reply:
I have changed the namespace accordingly